code/game/dna/mutations/mutation_powers.dm

```diff
@@ -385,13 +385,22 @@
 	if(!.)
 		return
 	var/can_eat = TRUE
-	if(iscarbon(user))
-		var/mob/living/carbon/C = user
-		if((C.head && (C.head.flags_cover & HEADCOVERSMOUTH)) || (C.wear_mask && (C.wear_mask.flags_cover & MASKCOVERSMOUTH) && !C.wear_mask.up))
-			if(show_message)
-				to_chat(C, "<span class='warning'>Your mouth is covered, preventing you from eating!</span>")
-			can_eat = FALSE
-	return can_eat
+	if(!iscarbon(user))
+		return can_eat
+
+	var/mob/living/carbon/C = user
+	if(!(C.head?.flags_cover & HEADCOVERSMOUTH))
+		return can_eat
+	if(!ismask(C.wear_mask))
+		return can_eat
+
+	var/obj/item/clothing/mask/worn_mask = C.wear_mask
+	if(!(worn_mask.flags_cover & MASKCOVERSMOUTH) || worn_mask.up)
+		return can_eat
+
+	if(show_message)
+		to_chat(C, "<span class='warning'>Your mouth is covered, preventing you from eating!</span>")
+	return FALSE
```

Comment:
This logic is pretty broken; it'll only say you can't eat if you have a hat *and* mask covering your mouth.